functionmap.go

```diff
@@ -2,166 +2,31 @@ package onvif
-	// PTZ Configuration
-	AddPTZConfiguration:    &media.AddPTZConfigurationFunction{},
-	RemovePTZConfiguration: &media.RemovePTZConfigurationFunction{},
-}
-
-var Media2FunctionMap = map[string]Function{
-	GetProfiles:                &media2.GetProfilesFunction{},
-	GetAnalyticsConfigurations: &media2.GetAnalyticsConfigurationsFunction{},
-	AddConfiguration:           &media2.AddConfigurationFunction{},
-	RemoveConfiguration:        &media2.RemoveConfigurationFunction{},
-}
-
-var PTZFunctionMap = map[string]Function{
-	// PTZ Capabilities
-	GetNodes: &ptz.GetNodesFunction{},
-	GetNode:  &ptz.GetNodeFunction{},
-	// PTZ Configuration
-	GetConfigurations:       &ptz.GetConfigurationsFunction{},
-	GetConfiguration:        &ptz.GetConfigurationFunction{},
-	GetConfigurationOptions: &ptz.GetConfigurationOptionsFunction{},
-	SetConfiguration:        &ptz.SetConfigurationFunction{},
-	// PTZ Actuation
-	AbsoluteMove:   &ptz.AbsoluteMoveFunction{},
-	RelativeMove:   &ptz.RelativeMoveFunction{},
-	ContinuousMove: &ptz.ContinuousMoveFunction{},
-	Stop:           &ptz.StopFunction{},
-	GetStatus:      &ptz.GetStatusFunction{},
-	// PTZ Preset
-	SetPreset:    &ptz.SetPresetFunction{},
-	GetPresets:   &ptz.GetPresetsFunction{},
-	GotoPreset:   &ptz.GotoPresetFunction{},
-	RemovePreset: &ptz.RemovePresetFunction{},
-	// PTZ Home Position
-	GotoHomePosition: &ptz.GotoHomePositionFunction{},
-	SetHomePosition:  &ptz.SetHomePositionFunction{},
-	// PTZ Auxiliary Operations
-	SendAuxiliaryCommand: &ptz.SendAuxiliaryCommandFunction{},
-}
-
-var EventFunctionMap = map[string]Function{
-	GetEventProperties:          &event.GetEventPropertiesFunction{},
-	CreatePullPointSubscription: &event.CreatePullPointSubscriptionFunction{},
-	PullMessages:                &event.PullMessagesFunction{},
-	Unsubscribe:                 &event.UnsubscribeFunction{},
-	Subscribe:                   &event.SubscribeFunction{},
-	Renew:                       &event.RenewFunction{},
-}
-
-var AnalyticsFunctionMap = map[string]Function{
-	// Analytics Module
-	GetSupportedAnalyticsModules: &analytics.GetSupportedAnalyticsModulesFunction{},
-	GetAnalyticsModules:          &analytics.GetAnalyticsModulesFunction{},
-	CreateAnalyticsModules:       &analytics.CreateAnalyticsModulesFunction{},
-	DeleteAnalyticsModules:       &analytics.DeleteAnalyticsModulesFunction{},
-	GetAnalyticsModuleOptions:    &analytics.GetAnalyticsModuleOptionsFunction{},
-	ModifyAnalyticsModules:       &analytics.ModifyAnalyticsModulesFunction{},
-	// Rule
-	GetSupportedRules: &analytics.GetSupportedRulesFunction{},
-	GetRules:          &analytics.GetRulesFunction{},
-	CreateRules:       &analytics.CreateRulesFunction{},
-	DeleteRules:       &analytics.DeleteRulesFunction{},
-	GetRuleOptions:    &analytics.GetRuleOptionsFunction{},
-	ModifyRules:       &analytics.ModifyRulesFunction{},
-}
-
 func FunctionByServiceAndFunctionName(serviceName, functionName string) (Function, error) {
```

Comment:
I removed all of the mappings above, and they are now auto-generated in `mappings.go`. The only function left here is this one, and I have slimmed it down by reducing the duplicated code.